nntrainer/models/neuralnet.cpp

```diff
@@ -568,7 +601,21 @@ std::vector<float *> NeuralNetwork::inference(std::vector<float *> &input,
       input[idx], in_dim[idx].getDataLen() * sizeof(float), in_dim[idx], 0)));
   }
 
-  sharedConstTensors output_tensors = inference(input_tensors, false);
+  if (!label.empty()) {
+    sharedConstTensors label_tensors;
+    auto label_dim = getOutputDimension();
+    label_tensors.reserve(label.size());
+    for (unsigned int idx = 0; idx < label_dim.size(); idx++) {
+      label_dim[idx].batch(batch_size);
+      label_tensors.emplace_back(MAKE_SHARED_TENSOR(
+        Tensor::Map(label[idx], label_dim[idx].getDataLen() * sizeof(float),
+                    label_dim[idx], 0)));
+    }
+    output_tensors = inference(input_tensors, label_tensors, false);
+  } else {
+    output_tensors = inference(input_tensors, false);
```

Comment:
can let this be called with label_tensors as empty `sharedConstTensors`, and have only 1 call to inference.